deepin-wm-dbus/deepinwmfaker.cpp

```diff
@@ -0,0 +1,1387 @@
+}
+
+void DeepinWMFaker::SetTransientBackgroundForMonitor(const QString &uri, const QString &strMonitorName)
+{
+     int current = m_windowSystem->currentDesktop();
+
+     m_transientBackgroundUri = uri;
+     Q_EMIT WorkspaceBackgroundChangedForMonitor( current,strMonitorName,uri );
+}
+
+#ifndef DISABLE_DEEPIN_WM
+void DeepinWMFaker::ChangeCurrentWorkspaceBackground(const QString &uri)
+{
+    SetCurrentWorkspaceBackground(uri);
+}
+#endif // DISABLE_DEEPIN_WM
+
+int DeepinWMFaker::GetCurrentWorkspace() const
+{
+    return m_windowSystem->currentDesktop();
+}
+
+int DeepinWMFaker::WorkspaceCount() const
+{
+    return m_windowSystem->numberOfDesktops();
+}
+
+void DeepinWMFaker::SetCurrentWorkspace(const int index)
+{
+    // 切换工作区时关闭壁纸预览
+    quitTransientBackground();
+    m_windowSystem->setCurrentDesktop(index);
+}
+
+void DeepinWMFaker::NextWorkspace()
+{
+    // loopback support
+//    int current = m_windowSystem->currentDesktop();
+//    ++current < m_windowSystem->numberOfDesktops() ? current : loopback ? 0 : --current;
+//    SetCurrentWorkspace(current);
+
+   SetCurrentWorkspace(m_windowSystem->currentDesktop() + 1);
+}
+
+void DeepinWMFaker::PreviousWorkspace()
+{
+    // loopback support
+//    int current = m_windowSystem->currentDesktop();
+//    --current >= 0 ? current : loopback ? --(m_windowSystem->numberOfDesktops()) : 0;
+//    SetCurrentWorkspace(current);
+
+    SetCurrentWorkspace(m_windowSystem->currentDesktop() - 1);
+}
+
+/*!
+ * [ { "Id":"...", "Accels":["...", "..."] }, {...} ]
+ */
+QString DeepinWMFaker::GetAllAccels() const
+{
+    QJsonArray allAccelsArray;
+    for (auto it = AllDeepinWMKWinAccelsMap.constBegin(); it != AllDeepinWMKWinAccelsMap.constEnd(); ++it) {
+        QJsonObject accelObj;
+        accelObj.insert("Id", it.key());
+        accelObj.insert("Accels", QJsonArray::fromStringList(GetAccel(it.key())));
+        accelObj.insert("Default", QJsonArray::fromStringList(GetDefaultAccel(it.key())));
+        allAccelsArray.append(accelObj);
+    }
+
+    return QJsonDocument(allAccelsArray).toJson(QJsonDocument::Compact);
+}
+
+/*!
+ * \brief DeepinWMFaker::GetAccel
+ * \param id The deepin wm accel name
+ * \return
+ */
+QStringList DeepinWMFaker::GetAccel(const QString &id) const
+{
+    if (id.isEmpty()) {
+        return QStringList();
+    }
+
+    const QString &kId = AllDeepinWMKWinAccelsMap.value(id);
+    if (kId.isEmpty()) {
+        return QStringList();
+    }
+
+    const QList<QKeySequence> &seqList = m_globalAccel->globalShortcut(GlobalAccelComponentName, kId);
+    if (seqList.isEmpty()) {
+        return QStringList();
+    }
+
+    QStringList accelList;
+    for (const QKeySequence &seq : seqList) {
+        accelList.append(transToDaemonAccelStr(seq.toString()));
+    }
+
+    return accelList;
+}
+
+static QMap<QString, QList<QKeySequence>> getShoutcutListFromKDEConfigFile()
+{
+    // 认为系统配置文件中存储的快捷键为默认值
+    KConfig kglobalshortcutsrc("/etc/xdg/kglobalshortcutsrc");
```

Comment:
能否使用 KSharedConfig::openConfig 避免硬编码
https://api.kde.org/frameworks/kconfig/html/classKSharedConfig.html

Reply:
```suggestion
    KConfig kglobalshortcutsrc(KSharedConfig::openConfig(QStringLiteral("kglobalshortcutsrc"), KConfig::SimpleConfig));
```

可能是这样的（没实际尝试）。

以及如果要读的是我们安装的配置的话，是不是文件名也改一下更合适？

Reply:
安装的配置文件都修改为 `deepin-` 前缀的了，同时也修改了所有代码中 load 的路径/文件名。